tests/modules/metarefresh/lib/MetaLoaderTest.php

```diff
@@ -0,0 +1,221 @@
+    {
+
+        $src = array(
+            'src' =>  __DIR__ . '/metadata-sample.xml',
+            'filterFactory' => 'sspmod_metarefresh_CommonFilters::entityAttributeFactory',
+            'filterFactoryArgs' => array('http://macedir.org/entity-category',
+                'http://refeds.org/category/research-and-scholarship')
+
+        );
+        $loader = new sspmod_metarefresh_MetaLoader();
+        $loader->loadSource($src);
+
+        $entities = $loader->getMetadata();
+
+        // Only sample entity with the correct category attribute name and value
+        $this->verifyEntityPresent(true, $entities, 'https://carmenwiki.osu.edu/shibboleth');
+        //osu.edu has the refeds category under a different name
+        $this->verifyEntityPresent(false, $entities, 'urn:mace:incommon:osu.edu');
+        $this->verifyEntityPresent(false, $entities, 'https://idp.nuim.ie/idp/shibboleth');
+
+    }
+
+    /**
+     * Test load metadata file with function factory
+     *
+     */
+    public function testLoadMetadataFileWithAuthorityFilterFactory()
+    {
+
+        $src = array(
+            'src' =>  __DIR__ . '/metadata-sample.xml',
+            'filterFactory' => 'sspmod_metarefresh_CommonFilters::registeredAuthorityFilterFactory',
+            'filterFactoryArgs' => array('http://www.heanet.ie')
+        );
+        $loader = new sspmod_metarefresh_MetaLoader();
+        $loader->loadSource($src);
+
+        $entities = $loader->getMetadata();
+
+        // Only heanet stuff makes it through the filter
+        $this->verifyEntityPresent(false, $entities, 'urn:mace:incommon:osu.edu');
+        $this->verifyEntityPresent(false, $entities, 'https://carmenwiki.osu.edu/shibboleth');
+        $this->verifyEntityPresent(true, $entities, 'https://idp.nuim.ie/idp/shibboleth');
+
+    }
+
+    /**
+     * Ensure both SAML 2 and 1.1 entities can be filtered
+     */
+    public function testLoadMetadataFileWithAuthorityFilterFactoryOnSAML1_1()
+    {
+
+        $src = array(
+            'src' =>  __DIR__ . '/metadata-sample.xml',
+            'filterFactory' => 'sspmod_metarefresh_CommonFilters::registeredAuthorityFilterFactory',
+            'filterFactoryArgs' => array('https://incommon.org')
+        );
+        $loader = new sspmod_metarefresh_MetaLoader();
+        $loader->loadSource($src);
+
+        $entities = $loader->getMetadata();
+
+        // Incommon SAML 2 and 1.1. entities should make it through
+        $this->verifyEntityPresent(true, $entities, 'urn:mace:incommon:osu.edu');
+        $this->verifyEntityPresent(true, $entities, 'https://carmenwiki.osu.edu/shibboleth');
+        $this->verifyEntityPresent(true, $entities, 'https://auth.cs.serialssolutions.com/auth/Metadata/Shib');
+        $this->verifyEntityPresent(false, $entities, 'https://idp.nuim.ie/idp/shibboleth');
+
+    }
+
+    /**
+     * Assert that the metadata has or doesn't have the entityId
+     * @param $present
+     * @param $metadata
+     * @param $entityId
```

Comment:
I know this is just a private method, but would you mind completing the phpdoc block with the correct types for the parameters, as well as a description?